lisa/sut_orchestrator/qemu/platform.py

```diff
@@ -65,21 +68,29 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
         self.qemu_platform_runbook = self.runbook.get_extended_runbook(
             QemuPlatformSchema, type_name=QEMU
         )
-        self.libvirt_conn_str = "qemu:///system"
 
-        if self.qemu_platform_runbook.is_host_remote():
-            self.host_node = RemoteNode(schema.Node(name="qemu-host"), 0, "qemu-host")
+    def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
+        host = self.qemu_platform_runbook.host
+        if host.is_remote():
+            assert host.address
```

Comment:
This is external data. So, you should throw a proper exception (`LisaException`) with a proper error message.

(FYI: `assert` is just a debug and type system declaration. It is valid for the compiler to remove it, if it wants to.)